kolena/_experimental/object_detection/utils.py

```diff
@@ -177,3 +177,97 @@ def compute_confusion_matrix_plot(
     for actual_label in ordered_labels:
         matrix.append([confusion_matrix[actual_label][predicted_label] for predicted_label in ordered_labels])
     return ConfusionMatrix(title=plot_title, labels=ordered_labels, matrix=matrix)
+
+
+def _compute_sklearn_arrays_by_class(
+    all_matches: List[MulticlassInferenceMatches],
+) -> Tuple[np.ndarray, np.ndarray, Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+    y_true_by_label: Dict[str, np.ndarray] = {}
+    y_score_by_label: Dict[str, np.ndarray] = {}
+
+    labels: Set[str] = set()
+    for match in all_matches:
+        for _, bbox_inf in match.matched:
+            labels.add(bbox_inf.label)
+        for bbox_gt, _ in match.unmatched_gt:
+            labels.add(bbox_gt.label)
+        for bbox_inf in match.unmatched_inf:
+            labels.add(bbox_inf.label)
+
+    for label in labels:
+        filtered_matchings = [
+            InferenceMatches(
+                matched=[(gt, inf) for gt, inf in match.matched if gt.label == label],
+                unmatched_gt=[gt for gt, _ in match.unmatched_gt if gt.label == label],
+                unmatched_inf=[inf for inf in match.unmatched_inf if inf.label == label],
+            )
+            for match in all_matches
+        ]
+
+        y_true, y_score = _compute_sklearn_arrays(filtered_matchings)
+        y_true_by_label[label] = y_true
+        y_score_by_label[label] = y_score
+
+    y_true, y_score = _compute_sklearn_arrays(all_matches)
```

Comment:
Do we need to compute and return these? They are not used by the only caller of this function.